pkg/controller/registrysyncer/registrysyncer_test.go

```diff
@@ -525,8 +537,133 @@ func TestReconcile(t *testing.T) {
 				return nil
 			},
 		},
+		{
+			name: "old IS on api.ci is deleted",
+			request: types.NamespacedName{
+				Name:      "applyconfig:latest",
+				Namespace: "ci",
+			},
+			apiCIClient: fakeclient.NewFakeClient(applyconfigISTag.DeepCopy(), applyconfigISDeleted.DeepCopy()),
+			appCIClient: fakeclient.NewFakeClient(applyconfigISTagNewer.DeepCopy(), applyconfigIS.DeepCopy()),
+
+			verify: func(apiCIClient ctrlruntimeclient.Client, appCIClient ctrlruntimeclient.Client) error {
+				actualImageStreamImport := &imagev1.ImageStreamImport{}
+				if err := apiCIClient.Get(ctx, types.NamespacedName{Name: "applyconfig", Namespace: "ci"}, actualImageStreamImport); !apierrors.IsNotFound(err) {
+					t.Errorf("expected NotFound error did not occur and the actual error is: %v", err)
+				}
+
+				actualImageStream := &imagev1.ImageStream{}
+				if err := apiCIClient.Get(ctx, types.NamespacedName{Name: "applyconfig", Namespace: "ci"}, actualImageStream); err != nil {
```

Comment:
I don't get it. If we delete it, why does it still exist?

Reply:
It is actually not deleted. It has a deleteTimestamp.
We need to remove our finalizer and kubelet will delete it when there is no finalizer.

Reply:
Ah, it already has the deletion timestamp set when we create it and we only verify that the finalizer is abset. Could make that more explicit, as we don't care about the rest